### DISCUSSION TITLE
Translatable assets

### DISCUSSION THREAD

**Discussion by mark-wff** [craftcms/cms#13110] [Ideas] [answered]
Is it possible to make images translatable in different languages, just like other fields?

Translation works perfectly fine for all other field types.

However, in almost all of our projects, there is a requirement to use different images for another language or another page group. We are currently using a matrix field and placing a file field within it as a workaround.

It would be great if we could select the translation method for assets as well.

**Comment by brandonkelly** [comment, marked as answer]
Files themselves cannot be translated, however you could create a separate volume that stores translated versions of assets, and relate them together using an Assets field. Templates could then be smart about showing the proper file depending on the current site language.

**Comment by mark-wff** [comment]
I don't think this really solves my problem. For example, I have a page for a client who has divided their page into two target groups, which are then translated into 10 languages.

I can conveniently specify for all fields in which group or language the fields should be synchronized or for which fields each language/group needs its own value. This feature is really very good and has saved the client a lot of work when adding content.

The only issue is with the files, where one needs to take an extra step. If there were a smarter way to handle this, CraftCMS would be very complete for our requirements.

It's less about actually translating the images, and more about selecting different images for different languages/target groups.

**Reply by brandonkelly** [reply]
You could create multiple Assets fields, one for each target group, and then within the field layout settings, give them conditions that ensure only the correct Assets field will show up depending on the entry/site being edited.
